openquake/hazardlib/geo/surface/planar.py

```diff
@@ -90,14 +90,14 @@ def mesh(self):
         return Mesh(self.corner_lons, self.corner_lats, self.corner_depths)
 
     def __init__(self, strike, dip,
-                 top_left, top_right, bottom_right, bottom_left):
-        if not (top_left.depth == top_right.depth and
-                bottom_left.depth == bottom_right.depth):
-            raise ValueError("top and bottom edges must be parallel "
-                             "to the earth surface")
-
-        NodalPlane.check_dip(dip)
-        NodalPlane.check_strike(strike)
+                 top_left, top_right, bottom_right, bottom_left, check=True):
```

Comment:
I'm fine with this solution for the time being. I wonder if the error is due to numerical approximation and could be solved by checking that the absolute difference is below a threshold.

Reply:
Actually it is the first thing I tried but the threshold would be very high.